src/locate.js

```diff
@@ -3,16 +3,19 @@ export const makeSuffixLocator = (suffix = '.js') => {
     // This constructor throws on anything but absolute URLs.
     // TODO: Ensure this is powerless, or inline string-manipulation code.
     const url = new URL(scopedRef);
+    let path = url.pathname;
 
     // Translate trailing slash to an index reference.
-    const path = url.pathname.endsWith('/')
-      ? `${url.pathname}index`
-      : url.pathname;
+    if (path.endsWith('/')) {
+      path = `${path}index`;
+    }
 
     // Translate missing suffix to have one.
-    const sfxPath = path.endsWith(suffix) ? path : `${path}${suffix}`;
+    if (!path.endsWith(suffix)) {
```

Comment:
Mutable state is easier to read than `?` expressions? Hm. not for me.

Reply:
FWIW, I find the new one easier to read and understand. The "?:" form, to express that one branch is good "as is", seems no easier to understand than the extremely local update of the `path` variable. When there's no possibility of aliasing, the uses and updates are all textually very local and co-visible, and none of the variables are captured or accessed by less direct control flow, then their's no loss of reliable static reasoning compared to the other form. This was also my reasoning for proposing that Jessie objects could be mutable and mutated as long as no aliasing was yet possible.

Reply:
fair enough.

ML habits die hard, I guess. (haskell, ocaml, ...)